src/execution_plan/ops/op_delete.c

```diff
@@ -30,30 +30,33 @@ void DeleteEntities(OpDelete *op) {
 	// lock everything
 	QueryCtx_LockForCommit();
 
-	if(GraphContext_HasIndices(op->gc)) {
-		for(int i = 0; i < node_count; i++) {
-			Node *n = op->deleted_nodes + i;
-			GraphContext_DeleteNodeFromIndices(op->gc, n);
-		}
+	// removing duplicates
+	Node *nodes = op->deleted_nodes;
+	Node *distinct_nodes = array_new(Node, 1);
```

Comment:
Removing duplicates shouldn't be performed under lock, we're wasting crucial time by doing so, we must keep our locking time to a minimum